File: codigo/python/selection_sort.py
def selection_sort(lista):
    n = len(lista)
    for i in range(n-1):
        min_index = i
        for j in range(i + 1, n):
            if lista[j] < lista[min_index]:
                min_index = j
        if i != min_index:
            temp = lista[i]
            lista[i] = lista[min_index]
            lista[min_index] = temp
    return lista

File: codigo/c/selection_sort.c
#include "selection_sort.h"

void selection_sort(Lista* lista) {
    int n = lista_tamanho(lista);
    
    for (int i = 0; i < n; i++) {
        int min_index = i;
        for (int j = i + 1; j < n; j++) {
            if (lista_get(lista, j) < lista_get(lista, min_index)) {
                min_index = j;
            }
        }
        if (i != min_index) {
            int temp = lista_get(lista, i);
            lista_set(lista, i, lista_get(lista, min_index));
            lista_set(lista, min_index, temp);
        }
    }
}
